Index: xamarin-pkg-fsharp/src/FSharpSource.targets
===================================================================
--- xamarin-pkg-fsharp.orig/src/FSharpSource.targets	2015-07-27 11:16:02.273487352 +0100
+++ xamarin-pkg-fsharp/src/FSharpSource.targets	2015-07-27 11:16:40.869487371 +0100
@@ -959,8 +959,7 @@
   <PropertyGroup>
     <SolutionDir>$(FSharpSourcesRoot)\..\</SolutionDir>
   </PropertyGroup>
-  <Import Project="$(FSharpSourcesRoot)\..\.nuget\NuGet.targets" />
   <Target Name="BeforeBuild" BeforeTargets="Build">
-    <Exec Command="$(NuGetCommand) restore packages.config -PackagesDirectory packages" WorkingDirectory="$(FSharpSourcesRoot)\.."/>
+    <Exec Command="nuget restore packages.config -PackagesDirectory packages" WorkingDirectory="$(FSharpSourcesRoot)\.."/>
   </Target>
 </Project>
