Index: xamarin-pkg-fsharp/src/FSharpSource.targets
===================================================================
--- xamarin-pkg-fsharp.orig/src/FSharpSource.targets
+++ xamarin-pkg-fsharp/src/FSharpSource.targets
@@ -649,8 +649,7 @@ Some other NuGET monikers to support in
         <CompileBefore Remove="@(CompileBefore)"/>   
     </ItemGroup>
   </Target>
-  <Import Project="$(FSharpSourcesRoot)\..\.nuget\NuGet.targets" />
   <Target Name="BeforeBuild" BeforeTargets="Build">
-    <Exec Command="$(NuGetCommand) restore packages.config -PackagesDirectory packages -ConfigFile $(FSharpSourcesRoot)\..\.nuget\NuGet.Config" WorkingDirectory="$(FSharpSourcesRoot)\.."/>
+    <Exec Command="nuget restore packages.config -PackagesDirectory packages -ConfigFile $(FSharpSourcesRoot)\..\.nuget\NuGet.Config" WorkingDirectory="$(FSharpSourcesRoot)\.."/>
   </Target>
 </Project>
Index: xamarin-pkg-fsharp/Makefile.in
===================================================================
--- xamarin-pkg-fsharp.orig/Makefile.in
+++ xamarin-pkg-fsharp/Makefile.in
@@ -1,7 +1,7 @@
 include $(topsrcdir)config.make
 
 all clean install build build-proto all-monotouch-monodroid-xamarinmac all-monotouch-monodroid all-monotouch all-monodroid all-xamarinmac all-monotouch-xamarinmac:
-	MONO_ENV_OPTIONS=$(monoopts) mono .nuget/NuGet.exe restore packages.config -PackagesDirectory packages -ConfigFile .nuget/NuGet.Config
+	MONO_ENV_OPTIONS=$(monoopts) nuget restore packages.config -PackagesDirectory packages -ConfigFile .nuget/NuGet.Config
 	$(MAKE) -C src/fsharp $@
 
 dist:
