Index: xamarin-pkg-fsharp/src/fsharp/Makefile.in
===================================================================
--- xamarin-pkg-fsharp.orig/src/fsharp/Makefile.in	2015-04-27 21:36:10.330451799 +0100
+++ xamarin-pkg-fsharp/src/fsharp/Makefile.in	2015-04-27 21:36:10.326451798 +0100
@@ -35,13 +35,7 @@
 	$(MAKE) -C policy.2.3.FSharp.Core $@
 	$(MAKE) -C policy.3.3.FSharp.Core $@
 	$(MAKE) -C policy.4.3.FSharp.Core $@
-	$(MAKE) -C policy.2.0.FSharp.Core TargetFramework=net20 $@
-	$(MAKE) -C policy.2.3.FSharp.Core TargetFramework=net20 $@
-	$(MAKE) -C FSharp.Core TargetFramework=net20 $@
-	$(MAKE) -C FSharp.Core TargetFramework=monodroid $@
-	$(MAKE) -C FSharp.Core TargetFramework=monotouch $@
 	$(MAKE) -C FSharp.Core FSharpCoreBackVersion=3.0 TargetFramework=net40 $@
-	$(MAKE) -C FSharp.Core FSharpCoreBackVersion=3.0 TargetFramework=net20 $@
 ifeq ("$(pclenabled47)", "yes")
 	$(MAKE) -C FSharp.Core TargetFramework=portable47 $@
 	$(MAKE) -C FSharp.Core FSharpCoreBackVersion=3.0 TargetFramework=portable47 $@
Index: xamarin-pkg-fsharp/FSharp.PowerPack/FSharp.PowerPack.fsproj
===================================================================
--- xamarin-pkg-fsharp.orig/FSharp.PowerPack/FSharp.PowerPack.fsproj	2015-04-27 21:36:01.822451461 +0100
+++ xamarin-pkg-fsharp/FSharp.PowerPack/FSharp.PowerPack.fsproj	2015-04-27 21:36:28.582452522 +0100
@@ -15,7 +15,7 @@
     <ProjectGuid>{649FA588-F02E-457C-9FCF-87E46407481F}</ProjectGuid>
     <OutputType>Library</OutputType>
     <AssemblyName>FSharp.PowerPack</AssemblyName>
-    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
     <TreatWarningsAsErrors>
     </TreatWarningsAsErrors>
     <DocumentationFile>FSharp.PowerPack.xml</DocumentationFile>
@@ -114,4 +114,4 @@
   <Import Project="$(MSBuildExtensionsPath32)\FSharp\1.0\Microsoft.FSharp.Targets" Condition="!Exists('$(MSBuildBinPath)\Microsoft.Build.Tasks.v4.0.dll')" />
   <Import Project="$(MSBuildExtensionsPath32)\..\Microsoft F#\v4.0\Microsoft.FSharp.Targets" Condition="Exists('$(MSBuildBinPath)\Microsoft.Build.Tasks.v4.0.dll')" />
   <Import Condition="'$(TargetFramework)'=='Silverlight'" Project="$(MSBuildExtensionsPath32)\Microsoft\Silverlight\$(SilverlightVersion)\Microsoft.Silverlight.Common.targets"/>
-</Project>
\ No newline at end of file
+</Project>
Index: xamarin-pkg-fsharp/src/fsharp/FSharp.Build-proto/FSharp.Build-proto.fsproj
===================================================================
--- xamarin-pkg-fsharp.orig/src/fsharp/FSharp.Build-proto/FSharp.Build-proto.fsproj	2015-04-27 21:36:01.822451461 +0100
+++ xamarin-pkg-fsharp/src/fsharp/FSharp.Build-proto/FSharp.Build-proto.fsproj	2015-04-27 21:36:35.026452778 +0100
@@ -14,7 +14,7 @@
     <DefineConstants>BUILDING_WITH_LKG;$(DefineConstants)</DefineConstants>
     <BuildWith>LKG</BuildWith>
     <ProjectGuid>{D8BC791F-C1A9-49DC-9432-0F3090537555}</ProjectGuid>
-    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
   </PropertyGroup>
   <!-- MonoDevelop likes these here to recognise the configurations -->
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' " />
Index: xamarin-pkg-fsharp/src/fsharp/FSharp.Compiler-proto/FSharp.Compiler-proto.fsproj
===================================================================
--- xamarin-pkg-fsharp.orig/src/fsharp/FSharp.Compiler-proto/FSharp.Compiler-proto.fsproj	2015-04-27 21:36:01.822451461 +0100
+++ xamarin-pkg-fsharp/src/fsharp/FSharp.Compiler-proto/FSharp.Compiler-proto.fsproj	2015-04-27 21:36:43.938453131 +0100
@@ -17,7 +17,7 @@
     <BuildWith>LKG</BuildWith>
     <AllowCrossTargeting>true</AllowCrossTargeting>
     <OtherFlags>$(OtherFlags) --warnon:1182</OtherFlags>
-    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
   </PropertyGroup>
   <!-- MonoDevelop likes these here to recognise the configurations -->
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' " />
Index: xamarin-pkg-fsharp/src/fsharp/FSharp.Compiler.Interactive.Settings/FSharp.Compiler.Interactive.Settings.fsproj
===================================================================
--- xamarin-pkg-fsharp.orig/src/fsharp/FSharp.Compiler.Interactive.Settings/FSharp.Compiler.Interactive.Settings.fsproj	2015-04-27 21:36:01.822451461 +0100
+++ xamarin-pkg-fsharp/src/fsharp/FSharp.Compiler.Interactive.Settings/FSharp.Compiler.Interactive.Settings.fsproj	2015-04-27 21:36:46.714453242 +0100
@@ -10,7 +10,7 @@
     <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
     <ProjectGuid>{649FA588-F02E-457C-9FCF-87E46407481E}</ProjectGuid>
     <OutputType>Library</OutputType>
-    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
     <DefineConstants>EXTENSIONTYPING;$(DefineConstants)</DefineConstants>
     <AssemblyName>FSharp.Compiler.Interactive.Settings</AssemblyName>
     <FileAlignment>512</FileAlignment>
Index: xamarin-pkg-fsharp/src/fsharp/FSharp.Data.TypeProviders.DesignTime/FSharp.Data.TypeProviders.DesignTime.fsproj
===================================================================
--- xamarin-pkg-fsharp.orig/src/fsharp/FSharp.Data.TypeProviders.DesignTime/FSharp.Data.TypeProviders.DesignTime.fsproj	2015-04-27 21:36:01.822451461 +0100
+++ xamarin-pkg-fsharp/src/fsharp/FSharp.Data.TypeProviders.DesignTime/FSharp.Data.TypeProviders.DesignTime.fsproj	2015-04-27 21:36:41.506453035 +0100
@@ -13,7 +13,7 @@
     <AllowCrossTargeting>true</AllowCrossTargeting>
     <!-- This DLL always targets .NET 4.0, no matter what, because it is a type provider design-time DLL -->
     <!-- For now, in this project file, TargetRuntimeFrameworkForDesignTimeComponent should always be set to some non-.NET-4.0 platform -->
-    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
     <ProjectGuid>{cb7d20c4-6506-406d-9144-5342c3595f03}</ProjectGuid>
     <OtherFlags>$(OtherFlags) --warnon:1182</OtherFlags>
   </PropertyGroup>
@@ -52,4 +52,4 @@
   <Import Project="$(FSharpSourcesRoot)\FSharpSource.targets" Condition=" !Exists('$(FSharpSourcesRoot)\FSharpTeam.targets') " />
   <Import Project="$(MSBuildExtensionsPath)\FSharp\1.0\Microsoft.FSharp.Targets" Condition=" !Exists('$(FSharpSourcesRoot)\FSharpTeam.targets') " />
   <Import Project="$(FSharpSourcesRoot)\..\lkg\FSharp-$(LkgVersion)\bin\FSharp.SRGen.targets" Condition=" !Exists('$(FSharpSourcesRoot)\FSharpTeam.targets') " />
-</Project>
\ No newline at end of file
+</Project>
Index: xamarin-pkg-fsharp/src/fsharp/FSharp.Data.TypeProviders/FSharp.Data.TypeProviders.fsproj
===================================================================
--- xamarin-pkg-fsharp.orig/src/fsharp/FSharp.Data.TypeProviders/FSharp.Data.TypeProviders.fsproj	2015-04-27 21:36:01.822451461 +0100
+++ xamarin-pkg-fsharp/src/fsharp/FSharp.Data.TypeProviders/FSharp.Data.TypeProviders.fsproj	2015-04-27 21:36:50.354453386 +0100
@@ -11,7 +11,7 @@
     <OutputType>Library</OutputType>
     <AssemblyName>FSharp.Data.TypeProviders</AssemblyName>
     <AllowCrossTargeting>true</AllowCrossTargeting>
-    <TargetFrameworkVersion Condition=" '$(TargetFrameworkVersion)' == '' ">v4.0</TargetFrameworkVersion>
+    <TargetFrameworkVersion Condition=" '$(TargetFrameworkVersion)' == '' ">v4.5</TargetFrameworkVersion>
     <ProjectGuid>{cb7d20c4-6506-406d-9144-5342c3595f03}</ProjectGuid>
     <DefineConstants>TPEMIT_INTERNAL_AND_MINIMAL_FOR_TYPE_CONTAINERS;$(DefineConstants)</DefineConstants>
    </PropertyGroup>
Index: xamarin-pkg-fsharp/src/fsharp/Fsc-proto/Fsc-proto.fsproj
===================================================================
--- xamarin-pkg-fsharp.orig/src/fsharp/Fsc-proto/Fsc-proto.fsproj	2015-04-27 21:36:01.822451461 +0100
+++ xamarin-pkg-fsharp/src/fsharp/Fsc-proto/Fsc-proto.fsproj	2015-04-27 21:36:38.698452924 +0100
@@ -20,7 +20,7 @@
     <ProjectGuid>{9D7C9060-9263-40EB-8FE3-1E4E3C6D941C}</ProjectGuid>
     <AllowCrossTargeting>true</AllowCrossTargeting>
     <OtherFlags>$(OtherFlags) --stackReserveSize:4096000</OtherFlags>
-    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
   </PropertyGroup>
   <!-- MonoDevelop likes these here to recognise the configurations -->
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' " />
