--- a/src/fsharp/Makefile.in
+++ b/src/fsharp/Makefile.in
@@ -35,13 +35,7 @@
 	$(MAKE) -C policy.2.3.FSharp.Core $@
 	$(MAKE) -C policy.3.3.FSharp.Core $@
 	$(MAKE) -C policy.4.3.FSharp.Core $@
-	$(MAKE) -C policy.2.0.FSharp.Core TargetFramework=net20 $@
-	$(MAKE) -C policy.2.3.FSharp.Core TargetFramework=net20 $@
-	$(MAKE) -C FSharp.Core TargetFramework=net20 $@
-	$(MAKE) -C FSharp.Core TargetFramework=monodroid $@
-	$(MAKE) -C FSharp.Core TargetFramework=monotouch $@
 	$(MAKE) -C FSharp.Core FSharpCoreBackVersion=3.0 TargetFramework=net40 $@
-	$(MAKE) -C FSharp.Core FSharpCoreBackVersion=3.0 TargetFramework=net20 $@
 ifeq ("$(pclenabled47)", "yes")
 	$(MAKE) -C FSharp.Core TargetFramework=portable-net45+sl5+win8 $@
 	$(MAKE) -C FSharp.Core FSharpCoreBackVersion=3.0 TargetFramework=portable-net45+sl5+win8 $@
