Index: xamarin-pkg-fsharp/launcher.in
===================================================================
--- xamarin-pkg-fsharp.orig/launcher.in
+++ xamarin-pkg-fsharp/launcher.in
@@ -21,4 +21,4 @@ fi
 # location of the default FSharp install in order to find the FSharp compiler binaries (see 
 # fsharpbinding/MonoDevelop.FSharpBinding/Services/CompilerLocationUtils.fs). That's a pretty unfortunate
 # way of finding those binaries. And really should be changed.
-$EXEC @MONOBINDIR@/mono $DEBUG $MONO_OPTIONS @DIR@/@TOOL@ --exename:$(basename "$0") "$@"
+SILLY_ENV_HACK_FOR_REGEX=mono $EXEC @MONOBINDIR@/mono $DEBUG $MONO_OPTIONS @DIR@/@TOOL@ --exename:$(basename "$0") "$@"
Index: xamarin-pkg-fsharp/src/fsharp/targets.make
===================================================================
--- xamarin-pkg-fsharp.orig/src/fsharp/targets.make
+++ xamarin-pkg-fsharp/src/fsharp/targets.make
@@ -213,7 +213,7 @@ install-sdk-lib:
 # This also installs 'fsharpc' and 'fsharpi'
 install-bin:
 	chmod +x $(outdir)$(ASSEMBLY)
-	sed -e 's,[@]DIR[@],$(gacdir)/fsharp,g' -e 's,[@]TOOL[@],$(ASSEMBLY),g' -e 's,[@]MONOBINDIR[@],$(monobindir),g' < $(topdir)launcher > $(outdir)$(subst fs,fsharp,$(NAME))
+	sed -e 's,[@]DIR[@],/usr/lib/cli/fsharp,g' -e 's,[@]TOOL[@],$(ASSEMBLY),g' -e 's,[@]MONOBINDIR[@],$(monobindir),g' < $(topdir)launcher > $(outdir)$(subst fs,fsharp,$(NAME))
 	chmod +x $(outdir)$(subst fs,fsharp,$(NAME))
 	@mkdir -p $(DESTDIR)$(gacdir)/fsharp
 	@mkdir -p $(DESTDIR)$(bindir)
